From fb847af696b873750adf2e61410d897d128f3dee Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolan.mcnamara@collabora.com>
Date: Sat, 12 Jul 2025 21:21:48 +0100
Subject: [PATCH] use const for these arrays

so they don't appear in the .data section. With const WordEndBits
ends up in the .rodata section and Formats in the .data.rel.ro
section.
---
 dict-generate.cpp | 2 +-
 zxcvbn.c          | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/dict-generate.cpp b/dict-generate.cpp
index 878b4aa..7041be7 100644
--- a/dict-generate.cpp
+++ b/dict-generate.cpp
@@ -1246,7 +1246,7 @@ int OutputCode(ostream *Out, bool Cmnts, const string & CharSet, StringIntSet_t
     unsigned int Len = ((NodeData.size() + 7) / 8);
     OutputSize += Len;
     x = 999;
-    *Out << "static unsigned char WordEndBits[" << Len << "] =\n{";
+    *Out << "static const unsigned char WordEndBits[" << Len << "] =\n{";
     Index = 0;
     unsigned int v = 0;
     unsigned int y = 0;
diff --git a/zxcvbn.c b/zxcvbn.c
index 95aa88f..f6d843b 100644
--- a/zxcvbn.c
+++ b/zxcvbn.c
@@ -1228,7 +1228,7 @@ static void SpatialMatch(ZxcMatch_t **Result, const uint8_t *Passwd, int Start,
 
 /* The possible date formats ordered by length (d for day, m for month, */
 /*  y for year, ? for separator) */
-static const char *Formats[] =
+static const char * const Formats[] =
 {
     "yyyy",
     "d?m?yy",
-- 
2.49.0

